src/core/gltf2importer/textureparser.cpp

```diff
@@ -68,34 +112,59 @@ bool TextureParser::parse(const QJsonArray &texturesArray, GLTF2ContextPrivate *
                 sourceValue = ddsExtension[KEY_SOURCE];
             }
         }
+
         if (sourceValue.isUndefined()) {
             qCWarning(kuesa, "Unknown image source for texture");
         } else {
             const auto image = context->image(sourceValue.toInt());
-            if (image.url.isEmpty())
+            if (image.url.isEmpty() && image.data.isEmpty())
                 return false; // Not a valid image
 
             auto texture2d = std::unique_ptr<Qt3DRender::QAbstractTexture>(nullptr);
-            if (isDDSTexture) {
-                auto textureLoader = new Qt3DRender::QTextureLoader;
-                texture2d.reset(textureLoader);
-                textureLoader->setSource(image.url);
+            if (image.data.isEmpty()) {
+                if (isDDSTexture) {
+                    auto textureLoader = new Qt3DRender::QTextureLoader;
+                    texture2d.reset(textureLoader);
+                    textureLoader->setSource(image.url);
+                } else {
+                    texture2d.reset(new Qt3DRender::QTexture2D);
+
+                    auto *textureImage = sharedImages.value(image.url);
+                    if (textureImage == nullptr) {
+                        auto ti = new Qt3DRender::QTextureImage();
+                        ti->setSource(image.url);
+                        ti->setMirrored(false);
+                        textureImage = ti;
+                        sharedImages.insert(image.url, textureImage);
+                    }
+
+                    // Add Image to Texture if compatible
+                    if (ensureImageIsCompatibleWithTexture(textureImage, texture2d.get()))
+                        texture2d->addTextureImage(textureImage);
+                    else
+                        qCWarning(kuesa) << "Image with source" << image.url << "is incompatbile with texture" << texture2d->objectName();
+                }
             } else {
-                texture2d.reset(new Qt3DRender::QTexture2D);
-
-                auto *textureImage = sharedImages.value(image.url);
-                if (textureImage == nullptr) {
-                    textureImage = new Qt3DRender::QTextureImage();
-                    textureImage->setSource(image.url);
-                    textureImage->setMirrored(false);
-                    sharedImages.insert(image.url, textureImage);
+                if (isDDSTexture) {
+                    // embedded DDS images not supported for now
```

Comment:
Maybe add a qWarning() saying this is not yet supported? So that users trying this at least get some feedback